e3db/src/main/java/com/tozny/e3db/Realm.kt

```diff
@@ -82,8 +80,44 @@ class Realm @JvmOverloads constructor(realmName: String?, appName: String?, brok
               JacksonConverterFactory.create(mapper)
           )
         }
+
     identityClient = retrofitBuilder.build().create()
 
+    // All network calls are done on background threads
+    // This is finicky, and does not guarantee that realmInfo will be populated before it is used?
+    backgroundExecutor.execute {
+      try {
+        val publicRealmInfo = identityClient.getPublicRealmInfo(realmName).execute()
+        when {
+          publicRealmInfo.isSuccessful -> {
+            publicRealmInfo.body().also {
+              if (it != null) {
+                this.realmInfo = it
+              }
+            }
+          }
+        }
+      } catch (e: Exception) {
+        // Cannot use uiExecutor.execute here similarly to Realm class methods because
+        // of no result handler
+        throw E3dbRealmNotFoundException(realmName)
+      }
+    }
+
+    // The above background thread execution, but done on main thread.
```

Comment:
Android won't allow network calls on the main thread so we need to do the call in the background.